pytorch_lightning/core/lightning.py

```diff
@@ -1703,13 +1703,14 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
             True
         """
 
-        if isinstance(input_sample, Tensor):
+        if input_sample is not None:
             input_data = input_sample
         elif self.example_input_array is not None:
             input_data = self.example_input_array
         else:
-            raise ValueError('`input_sample` and `example_input_array` tensors are both missing.')
-
+            raise ValueError('Could not export to ONNX since neither input_sample nor model.example_input_array '
+                             'attribute is set.')
```

Comment:
```suggestion
            raise ValueError('Could not export to ONNX since neither `input_sample` nor'
                             ' `model.example_input_array` attribute is set.')
```